### PR TITLE
CRUD: Setup

### PR DESCRIPTION
**Pull request by MoyasiGinko**
### In this milestone, I;

- Created a new branch `add-remove-btn`
- Created `data.js` file and the functionality of the page using CRUD
- Created a `clear.js` file for removing the tasks
- Updated `function.js` to edit tasks
- Created `storage.js` to update the localstorage
- Added tasks dynamically using javascript.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,5 +1,15 @@
...
-import renderTasks from './modules/data.js';
+const storage = new LocalStorage('tasks');
 
-window.addEventListener('load', renderTasks);
+const data = new Data();
+data.setTasks(storage.loadData());
+
+window.addEventListener('beforeunload', () => {
+  storage.saveData(data.getTasks());
+});
```
Comment (hiromataba) [inline src/index.js:11]: - [ ] By default, new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal to 5)

image

**Review by hiromataba — CHANGES_REQUESTED** [review]
Hi @MoyasiGinko    :wave: :wave: 

almost-there

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Your PR has a proper title and description  :+1: 
- Your linters are passing :heavy_check_mark:  
- Profesional README file :rocket: 

## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/index.js (hunk 1 above)
Comment (MoyasiGinko) [reply src/index.js:11]: Hello @hiromataba , The new task added to the array already has the value completed: false by default. Please check again.

**Review by mikethreels — APPROVED** [review]
Hi Moyasi, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job following the instructions of the previous CR.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**